src/content/en/updates/2017/02/media-session.md

```diff
@@ -0,0 +1,374 @@
+
+The cool thing about the Media Session API is that the notification tray is not
+the only place where media metadata and controls are visible. The media
+notification is synced automagically to any paired wearable device. And it also
+shows up on lock screens.
+
+<div class="clearfix"></div>
+<div class="attempt-left">
+  <figure>
+    <img src="https://placehold.it/350x350?text=Lock screen" alt="TODO">
+    <figcaption>Lockscreen</figcaption>
+  </figure>
+</div>
+<div class="attempt-right">
+  <figure>
+    <img src="https://placehold.it/350x350?text=Wear notification" alt="TODO">
+    <figcaption>Wear</figcaption>
+  </figure>
+</div>
+<div class="clearfix"></div>
+
+## Make it play nice offline
+
+I know what you're thinking now... *[Service Worker] to the rescue!*
+
+True but first and foremost, you want to make sure **all items in this
+checklist are checked**:
+
+1. All media and artwork files are served with the appropriate `Cache-Control`
+HTTP header. This will allow the browser to cache and reuse previously fetched
+resources. See the [Caching checklist].
+2. Make sure all media and artwork files are served with the
+`Allow-Control-Allow-Origin: *` HTTP header. This will allow third-party web
+apps to fetch and consume HTTP responses from your web server.
+
+### The Service Worker caching strategy
+
+Regarding media files, I would recommend a simple "[Cache, falling back to
+network]" strategy as illustrated by Jake Archibald.
+
+For artworks though, I'd be a little bit more specific and choose the approach
+below:
+
+- `If` artwork is already in cache, serve it from the cache
+- `Else` fetch artwork from network
+    - `If` fetch is successful, add network artwork to the cache and serve it
+    - `Else` serve fallback artwork from the cache
+
+That way, media notifications would always have a nice artwork icon even when
+browser can't fetch it. Here's how you could implement this for instance:
+
+    const FALLBACK_ARTWORK_URL = 'fallbackArtwork.png';
+    
+    self.addEventListener('install', event => {
+      self.skipWaiting();
+      event.waitUntil(initArtworksCache());
+    });
+    
+    function initArtworksCache() {
+      caches.open('artworks-cache-v1')
+      .then(cache => cache.add(FALLBACK_ARTWORK_URL));
+    }
+    
+    self.addEventListener('fetch', event => {
+      if (/artwork-[0-9]+\.png$/.test(event.request.url)) {
+        event.respondWith(handleFetchArtwork(event.request));
+      }
+    });
+    
+    function handleFetchArtwork(request) {
+      // Return cache request if it's in the cache already, otherwise fetch
+      // network artwork.
+      return getCacheArtwork(request)
+      .then(cacheResponse => cacheResponse || getNetworkArtwork(request));
+    }
+    
+    function getCacheArtwork(request) {
+      return caches.open('artworks-cache-v1')
+      .then(cache => cache.match(request));
+    }
+    
+    function getNetworkArtwork(request) {
+       // Fetch network artwork.
```

Comment:
It looks like this is a 3-space indent and the other functions have a 2-space indent.